compiler/utils/idioms.nim

```diff
@@ -33,7 +33,8 @@ func unreachableImpl(str: string, loc: IInfo) {.noinline, noreturn.} =
     if str.len > 0: " unreachable: "
     else:           " unreachable"
   msg.add str
-  raiseAssert(msg)
+  {.cast(noSideEffect).}:
```

Comment:
This is an internal compiler func, where are you running into issues with this one? I might be able to make it go away.